### ISSUE TITLE
Version string for both stable and pre-release NuGet builds 

### ISSUE THREAD

**Issue by asbjornu** [ParticularLabs/GitVersion#283]
I don't think GitVersion supports this scenario, so I'll just describe it and hope someone can point me in the right direction if this is already possible. If not, this can be considered a feature request.

I'd love to have a single build configuration across all branches of a project that ends up in a NuGet package. Stable releases should only be built from **master** and have `AssemblyInformationalVersion` and NuGet version set to:
* `{major}.{minor}.{patch}.{buildmetadata}`

All other branches (**development**, feature branches, etc.) should have `AssemblyInformationalVersion` and NuGet version set to:
* `{major}.{minor}.{patch}.{buildmetadata}-{branchname}`

The `-{branchname}` part will ensure that NuGet treats the package as a [prerelease](http://docs.nuget.org/docs/reference/versioning#Prerelease_Versions). This conditional switching on whether the `-{branchname}` part should be included (which it shouldn't on **master** builds), makes it difficult to just pass a variable to the "NuGet Pack" build step or have it use `AssemblyInformationalVersion`.

It seems like I should be able to use the `system.GitVersion.NuGetVersion` variable, but it seems to lack both `{buildmetadata}` and `{branchname}` (for both **master** and feature branches), so that won't work. `system.GitVersion.InformationalVersion` won't work either, since it contains `{fullbuildmetadata}` for both **master** and feature branch builds (which would cause **master** branch builds to end up as prerelease packages in NuGet).

So, I'm wondering if the variable I'm after exists. If it doesn't, I'd highly appreciate if it was added.

**Comment by JakeGinnivan** [comment]
@asbjornu hopefully this is sorted with the new config stuff. 

Could you retest using the release/3.0.0 branch (or I hopefully will publish 3.0.0-beta.1 today).

**Comment by JakeGinnivan** [comment]
At any rate, if 3.0.0 does not sort the issue. We need to reboot this issue as I can't really get anything actionable out of it at the moment

**Comment by asbjornu** [comment]
Brilliant! I'm going to give this a go with the GitVersionTask beta that is available now, as GitVersionTask is the solution I've found that gives most benefits with least amount of manual work. Being able to generate assemblies with the exact same version number across all computers, servers and builds, is phenomenal.

With version `2.0.1` of GitVersionTask, we're having some issues with people not being used to GitFlow and thus pushing commits to `master` that aren't tagged, causing GitVersionTask to yield no version number at all, which causes NuGet packages with version number `0.0.0.0` to be created.

I hope v3.0 be more resilient to manual errors like these. I'll let you know when we have tested.
